### DISCUSSION TITLE
base branch error when setting run on multiple branches

### DISCUSSION THREAD

**Discussion by sunafterrainwm** [renovatebot/renovate#20366] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Two base branches are set at [sunafterrainwm/LilyWhiteBot-typescript](https://github.com/sunafterrainwm/LilyWhiteBot-typescript), `master` and `develop`, renovate seems to have performed the update task of the two branches normally.
However, at [sunafterrainwm/LilyWhiteBot-typescript#9](https://github.com/sunafterrainwm/LilyWhiteBot-typescript/pull/9) , the base branch should be develop, but renovate mistakenly marks the base branch of pr as the default branch master.

### Relevant debug logs

https://gist.github.com/sunafterrainwm/4320d7f89b6f28a41a81ee226a50e83c

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment, marked as answer]
The repo in question has two base branches: https://github.com/sunafterrainwm/LilyWhiteBot-typescript/blob/72012efb60b8313edadb0a28454999498c845748/renovate.json#L6-L9

Lock file maintenance is enabled for both branches, and it seems to use the same branch name: `renovate/lock-file-maintenance`

Is this the first time lock file maintenance has run on the repo?

**Reply by sunafterrainwm** [reply]
Yes, this is the first time this repo has run lock file maintenance

This problem also occurs in another pr [sunafterrainwm/LilyWhiteBot-typescript#8](https://github.com/sunafterrainwm/LilyWhiteBot-typescript/pull/8) which is not lock file maintenance

**Reply by rarkins** [reply]
PR date:
<img width="495" alt="image" src="https://user-images.githubusercontent.com/6311784/219656473-fcc12268-c5b7-4843-82d1-cff9a829ca86.png">

Config update date:
<img width="640" alt="image" src="https://user-images.githubusercontent.com/6311784/219656539-491b4098-6e7c-462d-8618-a73591398d54.png">

So at the time of PR, your config had no baseBranches. After that Renovate seems to have been confused:

<img width="522" alt="image" src="https://user-images.githubusercontent.com/6311784/219656911-5a71474a-0912-4f14-8039-cb5f76819691.png">

Seems like there's possibly a bug, but triggered during the transition from no baseBranches to multiple baseBranches
